src/components/application_manager/test/commands/mobile/register_app_interface_request_test.cc

```diff
@@ -0,0 +1,323 @@
+        .WillByDefault(ReturnRef(kDummyString));
+    ON_CALL(app_mngr_settings_, supported_diag_modes())
+        .WillByDefault(ReturnRef(kDummyDiagModes));
+    ON_CALL(mock_policy_handler_, GetAppRequestTypes(_))
+        .WillByDefault(Return(std::vector<std::string>()));
+    ON_CALL(mock_policy_handler_, GetUserConsentForDevice(_))
+        .WillByDefault(Return(policy::DeviceConsent::kDeviceAllowed));
+    ON_CALL(app_mngr_, GetDeviceTransportType(_))
+        .WillByDefault(Return(hmi_apis::Common_TransportType::WIFI));
+    ON_CALL(app_mngr_, hmi_interfaces())
+        .WillByDefault(ReturnRef(mock_hmi_interfaces_));
+    ON_CALL(mock_hmi_interfaces_, GetInterfaceState(_))
+        .WillByDefault(Return(am::HmiInterfaces::STATE_NOT_AVAILABLE));
+    ON_CALL(
+        mock_hmi_interfaces_,
+        GetInterfaceFromFunction(hmi_apis::FunctionID::VR_ChangeRegistration))
+        .WillByDefault(Return(am::HmiInterfaces::HMI_INTERFACE_VR));
+    ON_CALL(
+        mock_hmi_interfaces_,
+        GetInterfaceFromFunction(hmi_apis::FunctionID::TTS_ChangeRegistration))
+        .WillByDefault(Return(am::HmiInterfaces::HMI_INTERFACE_TTS));
+    ON_CALL(
+        mock_hmi_interfaces_,
+        GetInterfaceFromFunction(hmi_apis::FunctionID::UI_ChangeRegistration))
+        .WillByDefault(Return(am::HmiInterfaces::HMI_INTERFACE_UI));
+  }
+
+  MessageSharedPtr msg_;
+  SharedPtr<RegisterAppInterfaceRequest> command_;
+
+  const utils::custom_string::CustomString app_name_;
+  sync_primitives::Lock lock_;
+  am::ApplicationSet app_set_;
+
+  typedef IsNiceMock<policy_test::MockPolicyHandlerInterface,
+                     kMocksAreNice>::Result MockPolicyHandlerInterface;
+
+  typedef IsNiceMock<resumprion_test::MockResumeCtrl, kMocksAreNice>::Result
+      MockResumeCtrl;
+
+  typedef IsNiceMock<connection_handler_test::MockConnectionHandler,
+                     kMocksAreNice>::Result MockConnectionHandler;
+
+  typedef IsNiceMock<protocol_handler_test::MockSessionObserver,
+                     kMocksAreNice>::Result MockSessionObserver;
+
+  typedef IsNiceMock<application_manager_test::MockHMICapabilities,
+                     kMocksAreNice>::Result MockHMICapabilities;
+
+  typedef IsNiceMock<am::MockHmiInterfaces, kMocksAreNice>::Result
+      MockHmiInterfaces;
+
+  MockPolicyHandlerInterface mock_policy_handler_;
+  MockResumeCtrl mock_resume_crt_;
+  MockConnectionHandler mock_connection_handler_;
+  MockSessionObserver mock_session_observer_;
+  MockHMICapabilities mock_hmi_capabilities_;
+  MockHmiInterfaces mock_hmi_interfaces_;
+};
+
+TEST_F(RegisterAppInterfaceRequestTest, Run_MinimalData_SUCCESS) {
+  InitGetters();
+  InitLanguage();
+  InitBasicMessage();
+
+  ON_CALL(app_mngr_, IsHMICooperating()).WillByDefault(Return(true));
+
+  MockAppPtr mock_app = CreateBasicMockedApp();
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(ApplicationSharedPtr()))
+      .WillRepeatedly(Return(mock_app));
```

Comment:
@pvvasilev, yes, but not completely. At least the `Init` part can be moved. But other call expectations cant be moved, because they can vary from one test case to another.